### PR TITLE
Remove hasteResolver in favor of standard esm relative imports [Fix #513]

### LINKED ISSUE #513: Replace `hasteResolvers` with relative imports
This project uses a custom "hasteResolver" defined in `build_helpers\resolvers.js` to allow simplified imports like `import FixedDataTableStore from 'FixedDataTableStore';` instead of relative imports like `import FixedDataTableStore from './FixedDataTableStore';`

While this works, as a newcomer to the project I found it unintuitive and confusing.   Also, my IDE can't understand the webpack resolution behavior, and is constantly warning me that it can't find these modules.

I'm curious about the decision not to use relative imports, and am wondering whether you'd be open to a PR to remove the custom resolver in favor of relative imports.

### PR DESCRIPTION
**Pull request by boxfoot**
## Description
Fixes #513 by removing the old "hasteResolvers" and using standard relative imports instead.

## Motivation and Context
The hasteResolvers is a legacy from when FB managed this package.  While it makes imports a little bit more concise, it obscures the architecture of the overall package and makes it harder for new contributors to understand how things fit together.  It's also a bit of magic that is only understand by webpack, making the code harder to maintain and reuse.  In contrast, relative imports are an old standard way of locating files on a file system and is totally supported by node.

Some notes/questions:

- Making this change emphases that there are a bunch of utilities and polyfills included under `vendor_upstream`, some of which are probably not needed anymore (like `requestAnimationFrame`, others that are duplicative (we sometimes use `vendor_upstream/core/clamp` and sometimes `lodash/clamp` and they both do the same thing) and others that could be easily provided by small npm packages.   I'm leaving making any of those changes out of scope for now, but we may want to address later.  For now, I've grouped imports into three groups in each file:  (1) raw package imports, (2) `vendor_upstream` imports, (3) local relative imports.
- Along the way I made some small changes to jsdoc/ts typings to silence errors I was getting in my IDE, and also added some `@types/` to the devDependencies for this purpose
- My code formatter also made some changes to whitespace, mostly making things more consistent within the repo (like removing invisible spaces) but if you don't want these, LMK and I can revert

## How Has This Been Tested?
Ran all tests, tried out examples.

## Types of changes

- [x] Non-feature refactor to simplify code

## Checklist:

- [x] My code follows the code style of this project.
- [x] ~My change requires a change to the documentation.~
- [x] ~I have updated the documentation accordingly.~
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by wcjordan** [comment]
Thanks for putting this PR up!  It looks like the tests are failing on travis-ci, have you had a chance to look into the cause of that yet?

File: src/vendor_upstream/stubs/EventListener.js [hunk 1]
```diff
@@ -25,18 +25,18 @@ var EventListener = {
    * @param {function} callback Callback function.
    * @return {object} Object with a `remove` method.
    */
-  listen: function(target, eventType, callback) {
+  listen: function (target, eventType, callback) {
```
Comment (wcjordan) [inline src/vendor_upstream/stubs/EventListener.js]: nit: don't add this space between the function keyword and the parameters.  Here and throughout the files.

**Comment by boxfoot** [comment]
@wcjordan thanks for the review!   I believe all addressed

>  It looks like the tests are failing on travis-ci

This was caused by unrelated changes pulled in from #507 that I hadn't noticed.  Now fixed.

> Thank you for the formatting fixes, they're much appreciated and needed. In future PRs could you separate the formatting changes to a different PR from the content so it's easier to review?

I can totally do in the future.  In this case, they weren't planned, just a side-effect of my IDE auto-formatting files as I worked on them.  (The various nits were also disagreements between my IDE's defaults and the standards in this repo).   Makes me wonder if it would be worth defining `prettierrc` and/or `eslintrc` for this repo and perhaps even hooking into commits with e.g. `husky` so that formatting is frictionless and automatic.

**Comment by wcjordan** [comment]
We definitely would like to have prettier run over the repo and a sane eslint would also be valuable as well.  There's some discussion on #218 & #455

**Comment by wcjordan** [comment]
@pradeepnschrodinger I'm ready to merge this given the formatting concerns are addressed by #520.  Do you have any concerns?  Should we wait until after v1.1 is released?

**Comment by wcjordan** [comment]
Let's hold off merging and get 1.1 released while we work the issues out w/ our NPM package.  Niranjan & I are discussing the NPM issues and will follow up here soon.

**Comment by boxfoot** [comment]
Sounds good.  LMK if there's something I missed or can fix to get the build back online.

**Comment by wcjordan** [comment]
Will do.  A lot of the issue has to do with how this [buildNPMInternals.sh|https://github.com/schrodinger/fixed-data-table-2/blob/master/build_helpers/buildNPMInternals.sh] script flattens the directory structure when preparing to upload to NPM.  We're discussing what we want the outcome to be, and will reach out once we know that (or in general details of why this old build system is how it is).

**Comment by wcjordan** [comment]
Hey @boxfoot sorry for our delays here.  We had some other priorities come up and haven't been able to investigate fixing buildNPMInternals.sh yet.    I just wanted to know we haven't forgotten about this, but it may be another month before we have a good answer.

**Comment by wcjordan** [comment]
Hey @boxfoot sorry again for all our delays here.  We've been heads down on some other projects, and are just now getting back to this.  We should be making progress over the coming few weeks.

**Comment by pradeepnschrodinger** [comment]
@boxfoot , hey really sorry for the delay. 
Our current build system uses a flat directory hierarchy, so it won't work with relative imports. Good thing is that if we just change how the build script arranges the files, the build works fine. 

So could you merge squash the commit 8cff56921701b78d73103f5e376baa98c05763b9 to this branch? The commit will update the build script to be compatible with relative imports, and then this PR would be good to go.

Also, there seems there's a few merge conflicts with latest master. Could you also take care of that?

**Comment by pradeepnschrodinger** [comment]
@boxfoot , merged this PR to master through 8677056dd9203db2b888b27394d9ffcb75788957. 
The follow up commit (1bcf8a2184eaab32426634ef9bb9b69102bf112c) fixes the build issues.

**Comment by boxfoot** [comment]
@pradeepnschrodinger Thanks for the update!

**Pull request merged**